chore: single binary without Kafka (#21318)
grafana/loki: docs/sources/shared/configuration.md

    # CLI flag: -dataobj-consumer.max-builder-age
    [max_builder_age: <duration> | default = 1h]

    # How records are ingested: "kafka" reads from a Kafka topic; "inmemory"
    # uses an in-process channel (experimental, single-node, no durability
    # guarantees, each replica holds independent data).
    # CLI flag: -dataobj-consumer.ingest-mode

>     [ingest_mode: <string> | default = "kafka"]

Reviewer: I would argue, that the default should be `inmemory`, because it is more likely to start a single binary (without Kafka) than a HA setup (with Kafka).

Author: I kept it default kafka to avoid changing the config in all existing clusters running on kafka mode already. Merging this should be no impact on them

Reviewer: I would really like to change the default to `inmemory` before we release this feature (after this has been merged and existing usages have been updated).

    # Buffered channel for in-process records (inmemory mode only). Each record
    # is ~100B-10KB depending on log line size. Default 10,000 is safe for most
    # experimental workloads. Monitor
    # loki_distributor_inmemory_dataobj_tee_stream_failures_total{reason="channel_full"}

>     # to alert on backpressure.

Reviewer: This help text is too technical and too in-depth.

```suggestion
    # Internal buffer size for records for `inmemory` ingestion.
    # Monitor
    loki_distributor_inmemory_dataobj_tee_stream_failures_total
    # to alert on backpressure.
```

  # to 0 to disable batching.
  # CLI flag: -distributor.dataobj-tee.rate-batch-window
  [rate_batch_window: <duration> | default = 0s]

> # Timeout for sending a record to the in-memory dataobj channel before returning

Reviewer: ```suggestion
# Timeout for sending a record to the in-memory queue before returning
```

> # backpressure to the caller. Default 5s matches the previous hardcoded value.

Reviewer: Changed by AI?

```suggestion
# backpressure to the caller.
```

Author: yep! That's claude :(
